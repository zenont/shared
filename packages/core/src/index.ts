export * from './common'
export * from './segregator'
export * from './util'
