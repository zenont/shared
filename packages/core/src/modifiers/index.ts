export * from './maybe'
export * from './immutable'
export * from './partial'
export * from './required'
export * from './promise'
