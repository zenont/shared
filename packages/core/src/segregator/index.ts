export * from './kind'
export * from './type'
