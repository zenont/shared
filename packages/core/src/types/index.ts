export * from './common'
export * from './segregator'
