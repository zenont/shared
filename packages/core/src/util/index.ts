export * from './env'
export * from './encoding'
