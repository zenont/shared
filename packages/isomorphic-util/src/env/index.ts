export * from './manager'
export * from './util'
export * from './types'
