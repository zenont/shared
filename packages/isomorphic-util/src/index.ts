import 'core-js/modules/esnext.global-this'
export * from './env'
export * from './encoding'
