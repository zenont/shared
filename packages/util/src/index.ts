export * from './env'
export * from './base64'
