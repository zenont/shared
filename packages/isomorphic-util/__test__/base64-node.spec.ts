/**
 * @jest-environment node
 */
import { toBase64, fromBase64 } from '../src'

const SYMBOLS_DECODED = '☸☹☺☻☼☾☿'
const SYMBOLS_ENCODED = '4pi44pi54pi64pi74pi84pi+4pi/'
const WALL_OF_TEXT = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum`
const ENCODED_WALL_OF_TEXT = `TG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQsIGNvbnNlY3RldHVyIGFkaXBpc2NpbmcgZWxpdCwgc2VkIGRvIGVpdXNtb2QgdGVtcG9yIGluY2lkaWR1bnQgdXQgbGFib3JlIGV0IGRvbG9yZSBtYWduYSBhbGlxdWEuIFV0IGVuaW0gYWQgbWluaW0gdmVuaWFtLCBxdWlzIG5vc3RydWQgZXhlcmNpdGF0aW9uIHVsbGFtY28gbGFib3JpcyBuaXNpIHV0IGFsaXF1aXAgZXggZWEgY29tbW9kbyBjb25zZXF1YXQuIER1aXMgYXV0ZSBpcnVyZSBkb2xvciBpbiByZXByZWhlbmRlcml0IGluIHZvbHVwdGF0ZSB2ZWxpdCBlc3NlIGNpbGx1bSBkb2xvcmUgZXUgZnVnaWF0IG51bGxhIHBhcmlhdHVyLiBFeGNlcHRldXIgc2ludCBvY2NhZWNhdCBjdXBpZGF0YXQgbm9uIHByb2lkZW50LCBzdW50IGluIGN1bHBhIHF1aSBvZmZpY2lhIGRlc2VydW50IG1vbGxpdCBhbmltIGlkIGVzdCBsYWJvcnVt`

describe('base64 node', () => {
  const ORIGINAL = process
  beforeEach(() => {
    jest.resetModules()
  })

  afterEach(() => {
    process = ORIGINAL
  })

  it('should encode utf-8 to base64', () => {
    const encoded = toBase64(WALL_OF_TEXT)
    expect(encoded).toBe(ENCODED_WALL_OF_TEXT)
  })

  it('should decode base64 to utf-8', () => {
    const decoded = fromBase64(ENCODED_WALL_OF_TEXT)
    expect(decoded).toBe(WALL_OF_TEXT)
  })

  it('should encode symbols utf-8 to base64', () => {
    const encoded = toBase64(SYMBOLS_DECODED)
    expect(encoded).toBe(SYMBOLS_ENCODED)
  })

  it('should decode symbols base64 to utf-8', () => {
    const decoded = fromBase64(SYMBOLS_ENCODED)
    expect(decoded).toBe(SYMBOLS_DECODED)
  })
})
