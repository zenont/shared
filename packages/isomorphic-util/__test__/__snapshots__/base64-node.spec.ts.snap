// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base64 node should encode symbols utf-8 to base64 1`] = `"4pi44pi54pi64pi74pi84pi+4pi/"`;
